docs/source/workflow-apis.md

```diff
@@ -0,0 +1,170 @@
+field.
+This is necessary so all the messages that are consumed and produced by the API can be picked up by the framework
+and the actor can depend on them.
+
+Here is an example of the usage:
+
+```python
+from leapp.actors import Actor
+from leapp.workflows.api.v2 import MyExampleAPI
+
+class MyWorkflowAPIUsingActor(Actor):
+    """ An example actor consuming the v2.MyExampleAPI Workflow API """
+    produces = ()
+    consumes = ()
+    apis = (MyExampleAPI,)
+
+    def process(self):
+        api = MyExampleAPI()
+        api.second(value=10)
+```
+
+## General workings of the Workflow API feature
+
+In the leapp repository , you want to create the API, for create a folder called `apis`.
+Now within the `apis` folder anything that is in there can be imported with the `leapp.workflows.api` prefix.
+An example: 
+
+You created a module `$repository/apis/example.py` this module can now be imported like this:
+
+```python
+from leapp.workflows.api import example
+```
+
+## Defining Workflow APIs
+
+A basic Workflow API is defined like this:
+
+```python
+from leapp.workflows.api import WorkflowAPI
+
+class MyExampleAPI(WorkflowAPI):
+    pass
+```
+
+This of course is no good as this wouldn't do anything just yet. APIs are nothing else than python classes and one
+just adds a method to it to make it available.
+
+Here is a simple function added to the API:
+
+```python
+from leapp.workflows.api import WorkflowAPI
+
+class MyExampleAPI(WorkflowAPI):
+    def first(self):
+        return 'First API function result'
+```
+
+### Working with messages
+
+Now if your API is going to consume or produce any messages, those messages have to be declared the same way as they
+are declared for actors.
+
+Let's assume we have defined two models: `Consumed` and `Produced` where the definitions look like this:
+```python
+from leapp.models import fields, Model
+from leapp.topics import ExampleTopic
+
+class Consumed(Model):
+    topic = ExampleTopic
+
+
+class Produced(Model):
+    topic = ExampleTopic
+    consumed = fields.Model(Consumed)
+    value = fields.Integer()
+```
+
+We are now going to define a method `second` which expects one parameter called value. That function 
+will produce a message of type `Produced` for each message of type `Consumed`. 
+
+```python
+from leapp.libraries.stdlib import api
+from leapp.messages import Consumed, Produced
+from leapp.workflows.api import WorkflowAPI
+
+class MyExampleAPI(WorkflowAPI):
+    consumes = (Consumed,)
+    produces = (Produced,)
+
+    def first(self):
+        return 'First API function result'
+
+    def second(self, value):
+        # Creates a new message `Produced` for each message of type `Consumed` with the additional value passed by 
+        # the caller.
+        for consumed in api.consume(Consumed):
+            self.produce(Produced(consumed=consumed, value=value))
+```
```

Comment:
Document writting of tests. To make it clear, that tests for apis are possible and recommended as well.